activemq-artemis/templates/configmap.yaml

```diff
@@ -17,33 +17,35 @@ data:
         {{- $fullname := include "artemis.fullname" . -}}
         {{- $releaseName := printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
         {{- $releaseNamespace := .Release.Namespace -}}
+
         <connectors>
           {{ range $i,$t := until (int .Values.replicas) }}
-            {{ range tuple "master" "slave" }}
-              <connector name="{{ $fullname }}-{{ . }}-{{ $i }}">tcp://{{ $fullname }}-{{ . }}-{{ $i }}.{{ $releaseName }}-{{ . }}.{{ $releaseNamespace }}.svc.cluster.local:61616</connector>  
-            {{ end }}
+          {{ range tuple "master" "slave" }}
+          <connector name="{{ $fullname }}-{{ . }}-{{ $i }}">tcp://{{ $fullname }}-{{ . }}-{{ $i }}.{{ $releaseName }}-{{ . }}.{{ $releaseNamespace }}.svc.cluster.local:61616</connector>
+          {{ end }}
           {{end}}
         </connectors>
 
+
         <cluster-connections>
           <cluster-connection name="{{ $releaseName }}">
             <address>jms</address>
-            <connector-ref>{{ $fullname }}-0</connector-ref>
+            <connector-ref>netty-connector</connector-ref>
             <retry-interval>500</retry-interval>
             <retry-interval-multiplier>1.1</retry-interval-multiplier>
             <max-retry-interval>5000</max-retry-interval>
             <initial-connect-attempts>-1</initial-connect-attempts>
             <reconnect-attempts>-1</reconnect-attempts>
 
-            <message-load-balancing>OFF</message-load-balancing>
+            <message-load-balancing>ON_DEMAND</message-load-balancing>
```

Comment:
Is this intentional?

Reply:
Yes, intentional. ON_DEMAND is the default. Assuming connections to our active nodes are distributed using the K8s load balancer, a Consumer attached to Node 1 won't receive a message from Producer attached to Node 2 unless this mode is activated.  Certainly for our deployment, we need additional node capacity to be available across the cluster.

Reply:
👍